src/tests/quantuminspire/test_api.py

```diff
@@ -513,20 +640,116 @@ def test_get_result_RaisesApiError(self):
+                                                       'list')
+        api = QuantumInspireAPI('FakeURL', self.authentication, coreapi_client_class=self.coreapi_client)
+        actual = api.get_result_from_job(identity)
+        self.assertDictEqual(actual, expected)
+
+    def test_get_result_from_job_RaisesApiError(self):
+        identity = 999
+        expected_payload = {'id': identity}
+        self.coreapi_client.handlers['jobs'] = partial(self.__mock_list_result_from_job_handler, expected_payload,
+                                                       'list')
+        api = QuantumInspireAPI('FakeURL', self.authentication, coreapi_client_class=self.coreapi_client)
+        self.assertRaises(ApiError, api.get_result_from_job, job_id=identity)
+
+    def test_get_raw_data_from_result_HasCorrectInputAndOutput(self):
+        identity = 485
+        expected_payload = {'id': identity, 'token': '162c'}
+        expected_raw_data = self.__mock_result_handler(expected_payload, 'read', None, None,
+                                                       ['test', 'raw-data', 'read'], expected_payload)
         self.coreapi_client.handlers['results'] = partial(self.__mock_result_handler, expected_payload, 'read')
-        self.coreapi_client.getters['https,//api.quantum-inspire.com/results/485/raw-data/162c/'] = \
-            self.__mock_raw_data_handler('https,//api.quantum-inspire.com/results/485/raw-data/162c/')
         api = QuantumInspireAPI('FakeURL', self.authentication, coreapi_client_class=self.coreapi_client)
-        actual = api.get_raw_data(result_id=identity)
+        actual = api.get_raw_data_from_result(result_id=identity)
         self.assertListEqual(actual, expected_raw_data)
 
+    def test_get_raw_data_unknown_from_result_RaisesApiError(self):
+        result_identity = 485
+        expected_payload = {'id': result_identity}
+        self.coreapi_client.handlers['results'] = partial(self.__mock_errors_in_result_handler,
+                                                          expected_payload, 'read')
+        api = QuantumInspireAPI('FakeURL', self.authentication, coreapi_client_class=self.coreapi_client)
+        self.assertRaisesRegex(ApiError, 'Invalid raw data url for result with', api.get_raw_data_from_result,
+                               result_id=result_identity)
+
+    def test_get_raw_data_invalid_from_result_RaisesApiError(self):
+        result_identity = 486
+        expected_payload = {'id': result_identity, 'token': '162c'}
+        self.coreapi_client.handlers['results'] = partial(self.__mock_errors_in_result_handler,
+                                                          expected_payload, 'read')
+        api = QuantumInspireAPI('FakeURL', self.authentication, coreapi_client_class=self.coreapi_client)
+        self.assertRaisesRegex(ApiError, 'Raw data for result with id', api.get_raw_data_from_result,
```

Comment:
Why leave the rest of the error message out? ...486 does not exist!

Reply:
Whole message added